src/main/java/com/game/racing/domain/car/Cars.java

```diff
@@ -0,0 +1,40 @@
+package com.game.racing.domain.car;
+
+import com.game.racing.domain.generator.NumberGenerator;
+import com.game.racing.view.ResultView;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Cars {
+
+    private final Map<String, Car> cars;
+
+    public Cars(String[] carNames, NumberGenerator numberGenerator) {
+        this.cars = new LinkedHashMap<>();
+        addNewCarsWithNames(carNames, numberGenerator);
+    }
+
+    private void addNewCarsWithNames(String[] carNames, NumberGenerator numberGenerator) {
+        for (String carName : carNames) {
+            this.cars.put(carName, new Car(carName, numberGenerator));
+        }
+    }
+
+    public Integer getTotalCarSize() {
+        return cars.size();
+    }
+
+    public void moveCars() {
+        for (String carName : cars.keySet()) {
+            cars.get(carName).move();
+        }
```

Comment:
```suggestion
        for (Map.Entry<String, Car> entry: cars.entrySet()) {
            entry.getValue().move();
        }
```
이런식으로 변경해볼수 있을듯 하네요.
아니면
```
for (Car car: cars.values()) {
            car.move();
        }
```
이런식도 있을꺼같습니다 :)

Reply:
좋은 의견 감사합니다. 해당 내용은 Map 객체에서 List 객체로 변경되어서 나중에 Map을 사용할 때 참고 해보겠습니다!